index.js

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env node
+require("dotenv").config();
+const axios = require("axios");
+
+// Usage from the command line:
+// npm install
+// node index sourceChannel targetChannel
+// e.g. node index general dev-chatter
+
+// Note: Do not include the hashtag in the channel name. Though I did account for stripping the hashtag away, process.argv returns undefined if the hashtag is present and I'm unsure of why.
+
+// Store Slack access token in a .env file in the root folder
+// You can ping a JDSB admin on Slack for the OAuth token, or request to become a collaborator on the app.
+// The token for an app can be found by selecting the app from api.slack.com/apps and navigating to "OAuth & Permissions" page after it has been installed to a workspace.
+const accessToken = process.env.SLACK_ACCESS_TOKEN;
+
+async function getConversationId(conversation) {
+
+  // Trim the pound sign if it was provided in the conversation argument
+  if (conversation.startsWith('#')) {
+    conversation = conversation.slice(1)
+  }
+
+  // Destructuring to pull out the channels result from the response
+  const { data: { channels } } = await axios.get(`https://slack.com/api/conversations.list?token=${accessToken}&pretty=1`)
+
+  // Find the channel with matching conversation name and pull out its ID
+  const conversationId = channels.find(chan => chan.name === conversation).id
+
+  return conversationId
+}
+
+async function getConversationMembers(conversationId) {
+
+  async function queryForMembers(membersCollection = [], nextCursor = '') {
+    let allConversationMembers = [];
+
+    // Base case: members exist, no next cursor, we've retrieved all the members
+    if (membersCollection.length > 0 && !nextCursor) {
+      return membersCollection;
+    }
+
+    try {
+      const response = await axios.get(
+        `https://slack.com/api/conversations.members?token=${accessToken}&channel=${conversationId}${nextCursor}`
+      );
+
+      // Looks ugly, but pulling out members and next_cursor
+      let {
+        data: {
+          members,
+          response_metadata: { next_cursor }
+        }
+      } = response;
+
+      // Cursor seems to always end with =
+      // Replace with URL encoding, %3D
+      if (next_cursor) {
+        next_cursor = `&cursor=${next_cursor.replace('=', '%3D')}`
```

Comment:
Are you sure this is the right thing to do? You should never modify cursor tokens...I'm surprised this didn't just break.

Reply:
This is where I should have known about `encodeURIComponent()` -- but essentially does the same thing.

As far as I could tell, next_cursor always ended with an equal sign, `=`, which needed to be encoded in the URL. It is in fact broke if I don't do this -- but maybe I missed something? I thought it was kind of odd as well.

Reply:
@bhurstGH good you found that encodeURIComponent() function. You rarely want to be hand-editing these things yourself.

Reply:
We'll make improvements for the Slack App migration.